tests/DPDK_Flow_Dump_with_action_selector.py

```diff
@@ -0,0 +1,156 @@
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to generate P4C artifacts or pb.bin")
+
+        # Create ports using gnmi-ctl
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to configure gnmi ctl ports")
+
+        # Set ip address for interface
+        ip_set_ipv4(self.interface_ip_list)
+
+        # Run Set-pipe command for set pipeline
+        if not p4rt_ctl.p4rt_ctl_set_pipe(
+            self.config_data["switch"],
+            self.config_data["pb_bin"],
+            self.config_data["p4_info"],
+        ):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to set pipe")
+
+        table = self.config_data["table"][0]
+
+        # Add action profile members
+        log.info("Add action profile members")
+        for member in table["member_details"]:
+            if not p4rt_ctl.p4rt_ctl_add_member_and_verify(
+                table["switch"], table["name"], member
+            ):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add member {member}")
+
+        # Adding action selector groups
+        log.info("Adding action selector groups")
+        group_count = 0
+        for group in table["group_details"]:
+            if not p4rt_ctl.p4rt_ctl_add_group_and_verify(
+                table["switch"], table["name"], group
+            ):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add group {group}")
+            group_count += 1
+
+        # Setting rule for l3 action selector
+        log.info(f"Setting up rule for : {table['description']}")
+        table = self.config_data["table"][1]
+        for match_action in table["match_action"]:
+            if not p4rt_ctl.p4rt_ctl_add_entry(
+                table["switch"], table["name"], match_action
+            ):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
+      
+        # Get dump entries table
+        dump_table = p4rt_ctl.p4rt_ctl_dump_entities(self.config_data['switch'])
+        # Verify each entry
+        if len(dump_table[1:-1]) != len(self.config_data["flow_dump_table"]):
+            log.failed(f"The table {dump_table[1:-1]} has different entry than definition")
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(f"Failed to verify flow dump entries\"")
+        else:
+            for each in dump_table[1:-1]:
+                entry = each.strip()
+                if entry in self.config_data["flow_dump_table"]:
+                    log.passed(f"The entry \"{entry}\" in flow dump is verified")
+                else:
+                    log.failed(f"The entry \"{entry}\" in flow dump is not verified")
```

Comment:
same as above

Reply:
The statements have a variables {entry}. The f is needed